feature/s3/manager/download_test.go

```diff
@@ -522,6 +550,73 @@ func TestDownload_WithRange(t *testing.T) {
 	}
 }
 
+func TestDownload_WithVersionID(t *testing.T) {
+	c, invocations, etags, versions := newDownloadVersionClient(buf12MB)
+
+	d := manager.NewDownloader(c)
+
+	w := manager.NewWriteAtBuffer(make([]byte, len(buf12MB)))
+	n, err := d.Download(context.Background(), w, &s3.GetObjectInput{
+		Bucket:    aws.String("bucket"),
+		Key:       aws.String("key"),
+		VersionId: aws.String("vid"),
+	})
+
+	if err != nil {
+		t.Fatalf("expect no error, got %v", err)
+	}
+	if e, a := int64(len(buf12MB)), n; e != a {
+		t.Errorf("expect %d buffer length, got %d", e, a)
+	}
+
+	if e, a := 3, *invocations; e != a {
+		t.Errorf("expect %v API calls, got %v", e, a)
+	}
+
+	expectVersions := []string{"vid", "vid", "vid"}
+	if e, a := expectVersions, *versions; !reflect.DeepEqual(e, a) {
+		t.Errorf("expect %v version ids, got %v", e, a)
+	}
+
+	expectETags := []string{"", "", ""}
```

Comment:
Not sure if the expected response is empty eTag, but this will likely still hold true: if a version id is set, the object should always have the same eTag

Reply:
`expectEtags` is list of `IfMatch` captured from requests, in this case versionId is passed by user so no ETag would be set to `IfMatch`